src/OpenLoco/Paint/PaintMiscEntity.cpp

```diff
@@ -109,7 +110,8 @@ namespace OpenLoco::Paint
             ImageIds::explosion_cloud_17
         };
 
-        uint32_t imageId = explosionCloudImageIds[particle->frame / 256];
+        assert((particle->frame / 256) >= 0 && static_cast<size_t>(particle->frame / 256) < explosionCloudImageIds.size());
```

Comment:
`particle->frame` should just be a uint, if it isn't already. You can then omit it from the assert. Other than that, this looks good to me.